pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +91,186 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
+	// Transform the spec minKubeVersion in semver Version to compare
+	if len(strings.Split(bundle.CSV.Spec.MinKubeVersion, ".")) == 2 {
+		bundle.CSV.Spec.MinKubeVersion += ".0"
+	}
+	semverMinKube, _ := semver.Make(bundle.CSV.Spec.MinKubeVersion)
+
+	// Check if the versions found/informed are that ones which the deprecated APIS are no longer supported (>= 1.22)
+	isUnsupported := semK8s.GE(k8s1_22) || semverMinKube.GE(k8s1_22)
+
+	// If the operator bundle is configured to support all k8s versions or any version >= 16
+	if len(k8sVersionFound) == 0 || semK8s.GE(k8s1_16) {
+		msg := fmt.Errorf("apiextensions/v1beta1 for CRDs and admissionregistration.k8s.io/v1beta1 for Webhooks was " +
+			"deprecated in Kubernetes `1.16` and removed in `1.22`. Use the version v1 instead")
+
+		// Note that if the operator bundle has both versions then we should only raise a warn and not an error since
+		// it might been covering both scenarios.
+		if hasDeprecatedAPIs(bundle) {
+			if isUnsupported && !hasV1(bundle) {
+				errs = append(errs, msg)
+			} else {
+				warns = append(errs, msg)
+			}
 		}
-		if maintainer.Email != "" {
-			_, err := mail.ParseAddress(maintainer.Email)
-			if err != nil {
-				errs = append(errs, fmt.Errorf("csv.Spec.Maintainers email %s is invalid: %v", maintainer.Email, err))
+	}
+
+	return errs, warns
+}
+
+// hasDeprecatedAPIs returns true when the bundle has one CRD/Webhook using v1beta1
+func hasDeprecatedAPIs(bundle *manifests.Bundle) bool {
+	if len(bundle.V1beta1CRDs) > 0 {
+		return true
+	}
+	for _, obj := range bundle.CSV.Spec.WebhookDefinitions {
+		for _, v := range obj.AdmissionReviewVersions {
+			if v == "v1beta1" {
+				return true
 			}
 		}
 	}
+	return false
+}
 
-	for _, link := range csv.Spec.Links {
-		if link.Name == "" || link.URL == "" {
-			errs = append(errs, fmt.Errorf("csv.Spec.Links elements should contain both name and url"))
+// hasV1 returns true when the bundle has a CRD/Webhook using V1
+func hasV1(bundle *manifests.Bundle) bool {
+	if len(bundle.V1CRDs) > 0 {
+		return true
+	}
+	for _, obj := range bundle.CSV.Spec.WebhookDefinitions {
+		for _, v := range obj.AdmissionReviewVersions {
+			if v == "v1" {
+				return true
+			}
 		}
-		if link.URL != "" {
-			_, err := url.ParseRequestURI(link.URL)
+	}
+	return false
+}
+
+// validateHubCSVSpec will check the CSV against the criteria to publish an
+// operator bundle in the OperatorHub.io
+func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+	var errs []error
+	var warns []error
+
+	errs = checkSpecProviderName(csv, errs)
+	errs = checkSpecMaintainers(csv, errs)
+	errs = checkSpecLinks(csv, errs)
+	errs, warns = checkAnnotations(csv, errs, warns)
+	errs = checkSpecVersion(csv, errs)
+	errs, warns = checkSpecIcon(csv, errs, warns)
+	errs, warns = checkSpecMinKubeVersion(csv, warns, errs)
+	errs, warns = checkBundleName(csv, warns, errs)
```

Comment:
See that all check still the same they are only encapsulated to reduce the complexity, be easier we cover then with the tests as to keep them maintained.